// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package mpis;

// The service definition.
service MPIService {
    rpc ISendRequest (RequestContext) returns (ReplyContext) {
    }
}

// The request message containing the user's name.
message RequestContext {
    // Rank of source (integer).
    int32 src = 1;
    // Rank of destination (integer).
    int32 dst = 2;
    // CPU or GPU
    int32 device = 3;
    // Number of elements in receive buffer (integer).
    int32 buffer_size = 4;
    // Message tag (integer).
    int32 tag = 5;
}

// The response message containing the greetings
message ReplyContext {
    // receive server is ready or not. 0 -> ready, else -> not ready
    int32 ready = 1;
    // if receive not ready, throw exception in message
    string msg = 2;
}
